proxy/src/auth/backend/jwt.rs

```diff
@@ -409,14 +409,22 @@ impl JwkCacheEntryLock {
 
         if let Some(exp) = payload.expiration {
             if now >= exp + CLOCK_SKEW_LEEWAY {
-                return Err(JwtError::InvalidClaims(JwtClaimsError::JwtTokenHasExpired));
+                return Err(JwtError::InvalidClaims(JwtClaimsError::JwtTokenHasExpired(
+                    exp.duration_since(SystemTime::UNIX_EPOCH)
```

Comment:
This made me realise we should use `checked_add` in the `numeric_date_opt` function.

Reply:
I'll change that too in this PR.

Reply:
I switched the deser function to use checked_add.